### PR TITLE
Rename stack-rook to provider-rook

### PR DESCRIPTION
**Pull request by hasheddan**
Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### Description of your changes

Changes references, imports, and module from `stack-rook` to `provider-rook`.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the dependencies in [`app.yaml`] to include any new role permissions.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`app.yaml`]: https://github.com/crossplane/stack-gcp/blob/master/config/stack/manifests/app.yaml

### CODE REVIEW

**Comment by hasheddan** [comment]
@negz don't forget about me! 😉

File: README.md [hunk 1]
```diff
@@ -1,53 +1,69 @@
...
+Conduct](https://github.com/crossplane/crossplane/blob/master/CODE_OF_CONDUCT.md)
+as the core Crossplane project.
 
 ## Licensing
 
-Stack-Rook is under the Apache 2.0 license.
+provider-rook is under the Apache 2.0 license.
 
-[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fcrossplane%2Fstack-rook.svg?type=large)](https://app.fossa.io/projects/git%2Bgithub.com%2Fcrossplane%2Fstack-rook?ref=badge_large)
+[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fcrossplane%2Fprovider-rook.svg?type=large)](https://app.fossa.io/projects/git%2Bgithub.com%2Fcrossplane%2Fprovider-rook?ref=badge_large)
```
Comment (jbw976) [inline README.md]: will we need to make updates on the FOSSA side or will that be done automatically?

File: config/stack/manifests/resources/yugabyteclusterclass.resource.yaml [hunk 2]
```diff
@@ -6,4 +6,4 @@ overviewShort: "A YugabyteClusterClass is a resource class. It defines the desir
 overview: |
   A YugabyteClusterClass is a resource class. It defines the desired spec of resource claims that use it to dynamically provision a managed resource.
 readme: |
-  Full documentation can be found in the Crossplane [API Reference docs](https://crossplane.io/docs/master/api/crossplane/stack-rook/storage-rook-crossplane-io-v1alpha1.html#YugabyteClusterClass).
```
Comment (jbw976) [inline config/stack/manifests/resources/yugabyteclusterclass.resource.yaml]: were the API docs already updated, or will those need to be completely regenerated for all repos in one big PR?

File: README.md (hunk 1 above)
Comment (hasheddan) [reply README.md]: I believe this will be done automatically :)

File: config/stack/manifests/resources/yugabyteclusterclass.resource.yaml (hunk 2 above)
Comment (hasheddan) [reply config/stack/manifests/resources/yugabyteclusterclass.resource.yaml]: they will need to be regenerated after all of the PR’s like this one are merged on each provider. I plan on those following this already open PR: https://github.com/crossplane/crossplane/pull/1322

**Pull request merged**